shell/plugins/steve/worker.js

```diff
@@ -0,0 +1,127 @@
+import * as Comlink from 'comlink';
+
+const COUNTS_FLUSH_TIMEOUT = 5000;
+const SCHEMA_FLUSH_TIMEOUT = 2500;
+
+const state = {
+  store:      '', // Store name
+  load:       undefined, // Load callback to load a resource into the store
+  counts:     [], // Buffer of count resources recieved in a given window
+  countTimer: undefined, // Tiemr to flush the count buffer
+  flushTimer: undefined, // Timer to flush the schema chaneg queue
+  queue:      [], // Schema change queue
+  schemas:    {} // Map of schema id to hash to track when a schema actually changes
+};
+
+// Quick, simple hash function
+function hash(str) {
+  let hash = 0;
+
+  for (let i = 0; i < str.length; i++) {
+    const char = str.charCodeAt(i);
+
+    hash = (hash << 5) - hash + char;
+    hash &= hash;
+  }
+
+  return new Uint32Array([hash])[0].toString(36);
+}
+
+// Quick, simple hash function to generate hash for an object
+function hashObj(obj) {
+  return hash(JSON.stringify(obj, null, 2));
+}
+
+function flush() {
+  state.queue.forEach((item) => {
+    const schema = item.schema;
+    const hash = hashObj(schema);
+    const existing = state.schemas[schema.id];
+
+    if (existing && existing !== hash) {
+      // console.log(`${ schema.id } CHANGED  ${ hash } > ${ existing }`);
+      state.schemas[schema.id] = hash;
+
+      const msg = {
+        data:          schema,
+        resoourceType: 'schema',
```

Comment:
`resoourceType` typo

Reply:
Done - don't think this is used - I think code looks at `data.type`